### PR TITLE
Add extraction service to CONFIG.md

### PR DESCRIPTION
**Pull request by navarone-feekery**
## Related to https://github.com/elastic/enterprise-search-team/issues/4579

Add a description for all local extraction service config

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [x] Considered corresponding documentation changes
- [x] Contributed any configuration settings changes to the configuration reference

### CODE REVIEW

File: docs/CONFIG.md [hunk 1]
```diff
@@ -38,6 +38,12 @@ Configuration lives in [config.yml](../config.yml).
   - `max_concurrent_access_control_syncs`: The maximum number of concurrent access control syncs. Defaults to 1.
   - `job_cleanup_interval`: The interval (in seconds) to run job cleanup task.
   - `log_level`: Connector service log level. Defaults to `INFO`.
+- `extraction_service`: Local extraction service-related configurations. These configurations are optional and are not included by default. The presence of these configurations enables local content extraction.
+  - `host`: The host of the local extraction service
```
Comment (navarone-feekery) [inline docs/CONFIG.md:42]:
```suggestion
  - `host`: The host of the local extraction service.
```

File: docs/CONFIG.md [hunk 2]
```diff
@@ -38,6 +38,12 @@ Configuration lives in [config.yml](../config.yml).
   - `max_concurrent_access_control_syncs`: The maximum number of concurrent access control syncs. Defaults to 1.
   - `job_cleanup_interval`: The interval (in seconds) to run job cleanup task.
   - `log_level`: Connector service log level. Defaults to `INFO`.
+- `extraction_service`: Local extraction service-related configurations. These configurations are optional and are not included by default. The presence of these configurations enables local content extraction.
+  - `host`: The host of the local extraction service.
+  - `timeout`: Request timeout for local extraction service requests, in seconds. Defaults to 30.
+  - `use_file_pointers`: Whether or not to use file pointers for local extraction. Defaults to `false`.
+  - `chunk_size`: The size that files are chunked to for streaming when sending a file to the local extraction service, in bytes. Only applicable if `use_file_pointers` is `false`. Defaults to 65536 (64KB).
```
Comment (navarone-feekery) [inline docs/CONFIG.md:45]:
```suggestion
  - `stream_chunk_size`: The size that files are chunked to for streaming when sending a file to the local extraction service, in bytes. Only applicable if `use_file_pointers` is `false`. Defaults to 65536 (64KB).
```

**Review by artem-shelkovnikov — APPROVED** [review]

File: docs/CONFIG.md [hunk 3]
```diff
@@ -38,6 +38,12 @@ Configuration lives in [config.yml](../config.yml).
   - `max_concurrent_access_control_syncs`: The maximum number of concurrent access control syncs. Defaults to 1.
   - `job_cleanup_interval`: The interval (in seconds) to run job cleanup task.
   - `log_level`: Connector service log level. Defaults to `INFO`.
+- `extraction_service`: Local extraction service-related configurations. These configurations are optional and are not included by default. The presence of these configurations enables local content extraction.
+  - `host`: The host of the local extraction service.
+  - `timeout`: Request timeout for local extraction service requests, in seconds. Defaults to 30.
+  - `use_file_pointers`: Whether or not to use file pointers for local extraction. Defaults to `false`.
+  - `stream_chunk_size`: The size that files are chunked to for streaming when sending a file to the local extraction service, in bytes. Only applicable if `use_file_pointers` is `false`. Defaults to 65536 (64KB).
+  - `shared_volume_dir`: The location for files to be extracted from. Only applicable if `use_file_pointers` is `true`. Defaults to `/app/files`.
```
Comment (artem-shelkovnikov) [inline docs/CONFIG.md:46]: Would it make sense to add more info here, mentioning how to set up Tika to also use this directory?

Comment (artem-shelkovnikov) [reply docs/CONFIG.md:46]: I think just a link is perfect :)

**Review by wangch079 — APPROVED** [review]

**Pull request merged**